homeassistant/components/shelly/__init__.py

```diff
@@ -376,7 +376,11 @@ def _async_device_updates_handler(self) -> None:
                     self.name,
                 )
 
-        if self._last_cfg_changed is not None and cfg_changed > self._last_cfg_changed:
+        if (
+            self._last_cfg_changed is not None 
+            and cfg_changed > self._last_cfg_changed 
+            and self.model not in DUAL_MODE_LIGHT_MODELS
+        ):
```

Comment:
This part is doing exactly the opposite from previous section:
https://github.com/home-assistant/core/blob/2a3a2e04af4225bfc14a64db85da4b83811ad98d/homeassistant/components/shelly/__init__.py#L331

Have you tried just the fixed in `light.py` that prevents setting mode if the bulb is already in the correct mode?
I prefer not to remove the functionality of reload unless it causes issues. For now the main problem is setting the mode when the bulb is already in this mode which increase the `cfgChanged`.

Reply:
I have tested without modification of __init__.py, it works flawlessy. There is no reload and unavailable status when switching mode.

Reply:
Thanks, If we encounter problems we can add it later